docs/testing.md

```diff
@@ -0,0 +1,140 @@
+## Testing
+
+`tsaga` comes with its testing utilities.
+
+The utilities are similar to `redux-saga-test-plan` and modeled after how a saga would run in your application.
+
+### Sagas
+
+The assertions use the `expect` function from the `jest` testing framework.
+
+```typescript
+import { expectSaga, select, call, run, spawn } from 'tsaga';
+
+it('should run a saga', () => {
+  return (
+    expectSaga(watchForFetchUser)
+      // Setup a store with a reducer and an optional initial state
+      // This is optional
+      .withReducer(rootReducer, undefined)
+      // Setup some mocks which should be used instead of the real functions
+      // The mock helper functions provide useful type assertions,
+      // so that you return a value which is valid
+      // This is optional
+      .withMocks([
+        // A select mock, the second argument is the return value when this mock is used
+        select(getUserById, { id: 5 }),
+        // A call mock, the second argument is the return value when this mock is used
+        call(fetchUserApi, { id: 5 }),
+        // A spawn mock, the second argument is the return value when this mock is used
+        spawn(trackUser),
+        // A run mock, the second argument is the return value when this mock is used
+        run(loginUser),
+      ])
+      // These are the expectations for the saga
+      // Expect the saga to call a function
+      // The arguments afterwards are what will be passed to the function
+      .toCall(fetchUserApi, 5, { all: true })
+      // Expect the saga to spawn a child bound function
+      .toSpawn(trackUser)
+      // Expect the saga to run a function
+      .toRun(loginUser, { id: 5 })
+      // Expect the saga to dispatch an action
+      .toDispatch(storeUser({ id: 5 }))
+      // Start the saga with the action
+      .dispatch(fetchUser(5))
+      // Do some assertions on the state
+      .toHaveFinalState()
+      // Start the whole saga and do the assertions
+      // You can provide a timeout as the first argument here for the saga to fail if it takes too long
+      // The default timeout is 10 seconds
+      .run()
+  );
+});
+```
+
+#### Mocks
+
+There are `select`, `call`, `run` and `spawn` mock helpers to easily setup mocks and have some type inference.
+These should only be used inside the tests and not in the actual saga code.
```

Comment:
Now that I read this I am wondering whether we should make this more obvious by export all test exports form `tsaga/testing` or something? (Not sure how that would mess up import on the consumer side though, it should never go to `tsaga/some/implementation/folder` instead).